test/com/amazon/ion/junit/IonAssert.java

```diff
@@ -112,12 +112,9 @@ public static void assertNoCurrentValue(IonReader in)
     @SuppressWarnings("deprecation")
     public static void assertEof(IonReader in)
     {
-        assertFalse(in.hasNext());
-        assertFalse(in.hasNext());
         assertNull(in.next());
         assertNoCurrentValue(in);
         assertNull(in.next());
-        assertFalse(in.hasNext());
         assertNull(in.next());
```

Comment:
Why repeat `assertNull(in.next())`? Does this deserve a comment?

Reply:
`IonReader.next()` may modify the internal state of the reader. Calling it twice at the end of a stream or container makes sure there isn't a bug in this logic that causes something weird to happen on successive calls. I've added a comment to call out that this repetition is intentional.

Reply:
Thanks!